### PR TITLE
Hint for shadowed std functions

### PR DESCRIPTION
**Pull request by Coekjan**
Related: #4263 

This PR adds a simple hint for shadowed std functions:

```typ
#let image = "image";

#image("ABC") // Hint here
```

We may need to hint for other shadowed things in the future, e.g. builtin variables, modules, so this PR shall not close that issue.

### CODE REVIEW

File: crates/typst-syntax/src/ast.rs [hunk 1]
```diff
@@ -1563,6 +1563,11 @@ node! {
 }
 
 impl<'a> FuncCall<'a> {
+    /// The function identifier.
+    pub fn ident(self) -> Ident<'a> {
```
Comment (laurmaedje) [inline crates/typst-syntax/src/ast.rs:1567]: The callee may not be an identifier, so this operation is a bit ill-defined. You should match on `callee` instead.

**Review by laurmaedje — COMMENTED** [review]
I wonder whether we can somehow solve this more generally, so that we don't have to spread this all over the place for functions, modules, etc. Seems tricky though.

**Comment by Coekjan** [comment]
> I wonder whether we can somehow solve this more generally, so that we don't have to spread this all over the place for functions, modules, etc. Seems tricky though.

Thanks for your feedback!

I believe that situations involving shadowed functions are more common and easier to trigger than those involving shadowed variables or modules. Therefore, I think we should prioritize providing hints for shadowed functions. After all, a general solution may not be as straightforward to conceive and implement.

File: crates/typst-syntax/src/ast.rs (hunk 1 above)
Comment (Coekjan) [reply crates/typst-syntax/src/ast.rs]: Thanks for the review.

I see `#"A"("ABC")` can be parsed to `FuncCall` and it has no identifier, so this operation is not good indeed. To make it better, I turned to let this function return `Option<Ident<'a>>` instead just now.

Comment (laurmaedje) [reply crates/typst-syntax/src/ast.rs]: why not just `match call.callee() { ast::Expr::Ident(ident) => .., _ => .. }` ?

Comment (Coekjan) [reply crates/typst-syntax/src/ast.rs]: Indeed. I've modified the code according to your suggestion.

**Comment by laurmaedje** [comment]
> Therefore, I think we should prioritize providing hints for shadowed functions.

Fair enough. Thanks!

File: crates/typst/src/eval/call.rs [hunk 2]
```diff
@@ -148,21 +148,21 @@ impl Eval for ast::FuncCall<'_> {
...
                 if let ast::Expr::Ident(ident) = self.callee() {
                     let ident = ident.get();
                     if vm.scopes.check_std_shadowed(ident) {
                         err.hint(eco_format!(
-                            "use `std.{}` to access the shadowed standard library function",
-                            ident,
-                        ));
+                        "use `std.{}` to access the shadowed standard library function",
+                        ident,
+                    ));
```
Comment (Coekjan) [inline crates/typst/src/eval/call.rs:159]: @laurmaedje indentations here seems broken :ghost:

**Pull request merged**